scrapy/downloadermiddlewares/retry.py

```diff
@@ -43,6 +43,7 @@ def get_retry_request(
     max_retry_times: Optional[int] = None,
     priority_adjust: Optional[int] = None,
     logger: Logger = retry_logger,
+    logger_give_up_level: int = 40,
```

Comment:
What about `give_up_log_level`?

Also, maybe we can make this `Union[int,str]`, to support the same API as the standard library, and use `"ERROR"` as default value, to make things more readable for people not aware of the value mappings.

Reply:
Regarding the feedback about "supporting the same API as the standard library" - the standard library does not support string arguments in the generic log function.

![image](https://user-images.githubusercontent.com/24959606/192247689-6a1808a4-87fd-4a92-872c-bcf714345033.png)

The library is written assuming that if you want to use labels as log level you should use a named method. The only way I've found to use a string as an argument is to have a mapping between the labels and their respective methods, but if a user is using a custom logger with different labels then I am not sure if we can create a mapping for them. Do you reckon it is okay to assume that everyone uses DEBUG, INFO, WARN, CRITICAL and ERROR; then leave the integer option for those who have added additional labels (and thus should be aware of their own mapping)?

Reply:
Oh, I did not know that. We could use [getLevelName](https://docs.python.org/3/library/logging.html#logging.getLevelName) to support custom log level strings as well. But I am also OK with forcing an int instead, given that is what the logging API supports. Whatever you prefer.